assets/app/features/upgrade/controllers/upgrade-upgrade-administration-server.controller.js

```diff
@@ -61,34 +78,24 @@
                 );
         }
 
-        function checkAdminUpgrade() {
-            upgradeFactory.getStatus()
-                .then(
-                    // In case of success
-                    function (response) {
-                        // map api response to model
-                        vm.adminUpgrade.completed = response.data.steps.admin_upgrade.status === 'passed';
-                        vm.adminUpgrade.running = response.data.steps.admin_upgrade.status === 'running';
-                    },
-                    // In case of failure
-                    function (errorResponse) {
-                        // Expose the error list to adminUpgrade object
-                        // TODO(itxaka): Use the proper error key from the response when this is done
-                        // https://trello.com/c/chzg85j4/142-3-s49p7-error-reporting-on-crowbar-level
-                        vm.adminUpgrade.errors = errorResponse.data.errors;
-                    }
-                ).finally(
-                    function () {
-                        // schedule another check if not completed yet
-                        if (!vm.adminUpgrade.completed && vm.adminUpgrade.running) {
-                            $timeout(vm.adminUpgrade.checkAdminUpgrade, ADMIN_UPGRADE_TIMEOUT_INTERVAL);
-                        }
-                        // If the upgrade was completed set the step to completed
-                        if (vm.adminUpgrade.completed) {
-                            upgradeStepsFactory.setCurrentStepCompleted();
-                        }
-                    }
-                );
+        function waitForUpgradeToEnd() {
+            upgradeStatusFactory.waitForStepToEnd(
+                UPGRADE_STEPS.admin_upgrade,
+                function (/*response*/) {
```

Comment:
Do you write a comment for each callback you pass to standard `when()`? Client code should not explain how library works. The library's docs are for that.